### ISSUE TITLE
incorrect currency exchange when minorDigits is 0

### ISSUE THREAD

**Issue by cjwind** [noojee/money.dart#25]
Hi, it seems that currency exchange is incorrect when currency's minorDigits is 0.

Example 1, TWD to USD:

```
Currency twd = Currency.create('TWD', 0, symbol: 'NT\$');
Currency usd = Currency.create('USD', 2);

Money twdM = Money.fromInt(1000, twd);
print(twdM);    // NT$1000.00

Money twdToUsdRate = Money.fromInt(3, usd);    // 1 TWD = 0.035 USD
print(twdToUsdRate);    // $0.03

Money usdM = twdM.exchangeTo(twdToUsdRate);
print(usdM);    // $0.30, but it should be $30
```

Example 2, TWD to JPY:

```
Currency jpy = Currency.create('JPY', 0, symbol: '¥');
Currency twd = Currency.create('TWD', 0, symbol: 'NT\$');

Money twdM = Money.fromInt(1000, twd);
print(twdM);    // NT$1000.00

Money twdToJpyRate = Money.fromInt(3, jpy);
print(twdToJpyRate);    // ¥3.00, 1 TWD = 3.68 JPY

Money jpyM = twdM.exchangeTo(twdToJpyRate);
print(jpyM);    // ¥30.00, but it should be ¥3000
```

There's another question: If we need to set more precise exchange rate, such as 1 TWD to 0.035582 USD, how could we set this rate through Money?

Thank you!

**Comment by bsutton** [comment]
I think the easiest way is to create your own version of the USD currency
with the required no of minor digits.

**Comment by cjwind** [comment]
Using my version currency to be precise exchange rate works. Thanks!

I think the exchange issue may caused by `~/ BigInt.from(100)` in `exchangeTo()`.
It seems to assume minorDigit of source currency is 2.

**Comment by bsutton** [comment]
I will check but I think there is a fix for that in the 2.0 version

**Comment by bsutton** [comment]
Closed. exchangeTo fixed in  2.0.1-nullsafety.7
